src/components/BuySellForm/BuySellForm.tsx

```diff
@@ -103,11 +103,11 @@ const BuySellForm = ({ advertId, isModalOpen, onRequestClose }: TBuySellFormProp
     const counterpartyRateRef = useRef<number | undefined>(undefined);
 
     useEffect(() => {
-        if (isAdvertiser) {
+        if (isAdvertiser || advertiserPaymentMethods === undefined) {
```

Comment:
will this work?

```suggestion
        if (isAdvertiser || !advertiserPaymentMethods) {
```

Reply:
because the advertiserPaymentMethods is supposed to return an empty object, this will return false if we negate it, so its a very rare edge case I have to check specifically if it becomes undefined, for some reason the whole data becomes undefined